### PR TITLE
Add Burstedde Weighted BFBT test

### PR DESCRIPTION
**Pull request by quangx**
This pull request adds a test using weighted BFBT (with Stokes solver type = block AMG).

### CODE REVIEW

File: tests/burstedde_bfbt.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env perl
+
```
Comment (tjhei) [inline tests/burstedde_bfbt.sh:2]: - [ ] Could you please remove this sh file and update test results?

**Review by tjhei — APPROVED** [review]

File: tests/burstedde_bfbt.prm [hunk 2]
```diff
@@ -0,0 +1,105 @@
+# The 3D polynomial Stokes (burstedde) benchmark, for which an
```
Comment (tjhei) [inline tests/burstedde_bfbt.prm:1]: can you change the comment to read "test burstedde benchmark using BFBT preconditioner", remove all the other comments here, include the burstedde.prm from the benchmarks folder and only include the relevant sections here (mesh refinement and solver settings).

**Review by tjhei — APPROVED** [review]

**Review by tjhei — APPROVED** [review]

**Comment by tjhei** [comment]
you will need to update test results once again

**Review by gassmoeller — APPROVED** [review]
I am ok with merging this to have a test for the bfbt schur complement preconditioner, but just to document this for the future: At the moment the bfbt version requires 159 outer GMRES iterations and the simple inverse viscosity mass matrix requires only 18 outer GMRES iterations. To compute this I used the same parameter file you have here inside the tester docker container, but with `set Use weighted BFBT for Schur complement          = false`. In other words either the method doesnt work very well for this benchmark, or there is something wrong in the method implementation. Did I interpret this correctly?

**Comment by quangx** [comment]
> I am ok with merging this to have a test for the bfbt schur complement preconditioner, but just to document this for the future: At the moment the bfbt version requires 159 outer GMRES iterations and the simple inverse viscosity mass matrix requires only 18 outer GMRES iterations. To compute this I used the same parameter file you have here inside the tester docker container, but with `set Use weighted BFBT for Schur complement = false`. In other words either the method doesnt work very well for this benchmark, or there is something wrong in the method implementation. Did I interpret this correctly?

From all the tests I have run, AMG-BFBT can take quite a lot of iterations to converge when the number of initial global refinements is small.

If you start with four initial global refinements, AMG takes 18 iterations and AMG-BFBT takes 73. For five levels,
AMG takes 19 and AMG-BFBT takes 65.

**Comment by gassmoeller** [comment]
Thanks for confirming. However, is this a state you would expect? From my understanding BFBT was supposed to be a *better* approximation to the real Schur complement than the inverse mass matrix and therefore should reduce the number of outer GMRES iterations. Did you find a theoretical reason why it is worse?

**Comment by tjhei** [comment]
Yes, we are aware. I asked him to open PRs to add tests so we can document the current status and can correctly track progress.

In short: we do not implement the corrections mentioned in the original paper for coarse meshes and I suspect our inner solver is not done correctly either.

**Pull request merged**